lib/std/os/linux.zig

```diff
@@ -1386,6 +1386,58 @@ pub fn madvise(address: [*]u8, len: usize, advice: u32) usize {
     return syscall3(.madvise, @ptrToInt(address), len, advice);
 }
 
+pub fn posix_fadvise(fd: fd_t, offset: off_t, len: off_t, advice: usize) usize {
+    if (usize_bits == 64 and @hasField(SYS, "fadvise64")) {
+        return syscall4(
+            .fadvise64,
```

Comment:
The idea is to follow musl's strategy of using `fadvise64_64` if available and then fall back unconditionally to `SYS_fadvise64`. This order ensures the syscall with fewer arguments is used when available.

Reply:
Addressed in the latest commit.